### ISSUE TITLE
details about the base model and the fine-tuned

### ISSUE THREAD

**Issue by futureisatyourhand** [bingykang/Fewshot_Detection#18]
I use two gpus, the other configurations is the same as the author, why is my performance poor?

- The following figure is the eval of 500 epochs for base model.
![image](https://github.com/futureisatyourhand/images/blob/master/500.png)

- The following figure is the eval of  5 epochs for the fine-tune.
![image](https://github.com/futureisatyourhand/images/blob/master/5.png)

- The following figure is the eval of  10 epochs for the fine-tune.
![image](https://github.com/futureisatyourhand/images/blob/master/10.png)

**Comment by cgg99** [comment]
@futureisatyourhand  Hi~How did you evaluate the base model？Did you execute `python valid_ensemble.py cfg/metayolo_novel0.data cfg/darknet_dynamic.cfg cfg/reweighting_net.cfg path/toweightfile` ？

**Comment by futureisatyourhand** [comment]
> @futureisatyourhand Hi~How did you evaluate the base model？Did you execute `python valid_ensemble.py cfg/metayolo_novel0.data cfg/darknet_dynamic.cfg cfg/reweighting_net.cfg path/toweightfile` ？

Hello, I have solved the problem, but the AP is much higher than the author's, the randomness is very strong.

**Comment by MonsterLWR** [comment]
i've met similar problem, could you please tell us how you solve it? @futureisatyourhand  
thanks!

**Comment by futureisatyourhand** [comment]
> i've met similar problem, could you please tell us how you solve it? @futureisatyourhand
> thanks!
First of all, the author's code is very random, the AP after each training is different, you'd better set up a random seed.
Second, the rest of the git code cannot be changed, you can try to retrain the git author's code, you may find that the results are better than the author's code, the author's code is very metaphysical, I just re-git the author's code and training results are better than the author.
For details, you can look at the results that I reproduced and optimized the algorithm, the linke is https://arxiv.org/pdf/2007.06837.pdf.

**Comment by MonsterLWR** [comment]
Thanks for your reply! I still have some questions:
how many gpus did you use? when using two gpus, should i change the learning rate or max batches?

**Comment by MonsterLWR** [comment]
Thanks for your reply! I still have some questions:
how many gpus did you use? when using two gpus, should i change the learning rate or max batches? @futureisatyourhand

**Comment by GDDKRJ** [comment]
Hi~How did you perform the code when evaluating the model？
“python valid_ensemble.py cfg/metayolo.data cfg/darknet_dynamic.cfg cfg/reweighting_net.cfg path/toweightfile
python scripts/voc_eval.py results/path/to/comp4_det_test_”
I don’t know the “path/toweightfile” and “results/path/to/comp4_det_test_”

**Comment by futureisatyourhand** [comment]
> Hi~How did you perform the code when evaluating the model？
> “python valid_ensemble.py cfg/metayolo.data cfg/darknet_dynamic.cfg cfg/reweighting_net.cfg path/toweightfile
> python scripts/voc_eval.py results/path/to/comp4_det_test_”
> I don’t know the “path/toweightfile” and “results/path/to/comp4_det_test_”

The weight file is your best weights saved, and the "results/path/to/comp4_det_test_" is your results containing boxes and label. when the first instruction executed successfully, and the the  "results/path/to/comp4_det_test_"  is generated. You should scan source code.

**Comment by GDDKRJ** [comment]
Is the first instruction written like this?&nbsp;python valid_ensemble.py cfg/metayolo.data cfg/darknet_dynamic.cfg cfg/reweighting_net.cfg backup/metayolo/000010.weights
I want to know the best weights is darknet19_448.cfg or backup/metayolo/****.weights?
Because when I use darknet19_448.cfg the code can't run


------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "bingykang/Fewshot_Detection"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年9月23日(星期四) 上午9:19
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [bingykang/Fewshot_Detection] details about the base model and the fine-tuned (#18)





  
Hi~How did you perform the code when evaluating the model？
 “python valid_ensemble.py cfg/metayolo.data cfg/darknet_dynamic.cfg cfg/reweighting_net.cfg path/toweightfile
 python scripts/voc_eval.py results/path/to/comp4_det_test_”
 I don’t know the “path/toweightfile” and “results/path/to/comp4_det_test_”
  
The weight file is your best weights saved, and the "results/path/to/comp4_det_test_" is your results containing boxes and label. when the first instruction executed successfully, and the the  "results/path/to/comp4_det_test_"  is generated. You should scan source code.
 
—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by futureisatyourhand** [comment]
> Is the first instruction written like this?&nbsp;python valid_ensemble.py cfg/metayolo.data cfg/darknet_dynamic.cfg cfg/reweighting_net.cfg backup/metayolo/000010.weights I want to know the best weights is darknet19_448.cfg or backup/metayolo/****.weights? Because when I use darknet19_448.cfg the code can't run
> […](#)
> ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: "bingykang/Fewshot_Detection" ***@***.***&gt;; 发送时间:&nbsp;2021年9月23日(星期四) 上午9:19 ***@***.***&gt;; ***@***.******@***.***&gt;; 主题:&nbsp;Re: [bingykang/Fewshot_Detection] details about the base model and the fine-tuned (#18) Hi~How did you perform the code when evaluating the model？ “python valid_ensemble.py cfg/metayolo.data cfg/darknet_dynamic.cfg cfg/reweighting_net.cfg path/toweightfile python scripts/voc_eval.py results/path/to/comp4_det_test_” I don’t know the “path/toweightfile” and “results/path/to/comp4_det_test_” The weight file is your best weights saved, and the "results/path/to/comp4_det_test_" is your results containing boxes and label. when the first instruction executed successfully, and the the "results/path/to/comp4_det_test_" is generated. You should scan source code. — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe. Triage notifications on the go with GitHub Mobile for iOS or Android.

the 



> Is the first instruction written like this?&nbsp;python valid_ensemble.py cfg/metayolo.data cfg/darknet_dynamic.cfg cfg/reweighting_net.cfg backup/metayolo/000010.weights I want to know the best weights is darknet19_448.cfg or backup/metayolo/****.weights? Because when I use darknet19_448.cfg the code can't run
> […](#)
> ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: "bingykang/Fewshot_Detection" ***@***.***&gt;; 发送时间:&nbsp;2021年9月23日(星期四) 上午9:19 ***@***.***&gt;; ***@***.******@***.***&gt;; 主题:&nbsp;Re: [bingykang/Fewshot_Detection] details about the base model and the fine-tuned (#18) Hi~How did you perform the code when evaluating the model？ “python valid_ensemble.py cfg/metayolo.data cfg/darknet_dynamic.cfg cfg/reweighting_net.cfg path/toweightfile python scripts/voc_eval.py results/path/to/comp4_det_test_” I don’t know the “path/toweightfile” and “results/path/to/comp4_det_test_” The weight file is your best weights saved, and the "results/path/to/comp4_det_test_" is your results containing boxes and label. when the first instruction executed successfully, and the the "results/path/to/comp4_det_test_" is generated. You should scan source code. — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe. Triage notifications on the go with GitHub Mobile for iOS or Android.

the instruction is to test, so it is no relation to  darknet19_448.cfg, the  darknet19_448.weights is pretrained weights to get backup/metayolo/000010.weights!!!

**Comment by GDDKRJ** [comment]
I want to know why the ***.txt is empty?It has no information.


------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "bingykang/Fewshot_Detection"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年9月23日(星期四) 中午11:40
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [bingykang/Fewshot_Detection] details about the base model and the fine-tuned (#18)





  
Is the first instruction written like this?&nbsp;python valid_ensemble.py cfg/metayolo.data cfg/darknet_dynamic.cfg cfg/reweighting_net.cfg backup/metayolo/000010.weights I want to know the best weights is darknet19_448.cfg or backup/metayolo/.weights? Because when I use darknet19_448.cfg the code can't run
 …
 ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: "bingykang/Fewshot_Detection" @.&gt;; 发送时间:&nbsp;2021年9月23日(星期四) 上午9:19 @.&gt;; @.@.**&gt;; 主题:&nbsp;Re: [bingykang/Fewshot_Detection] details about the base model and the fine-tuned (#18) Hi~How did you perform the code when evaluating the model？ “python valid_ensemble.py cfg/metayolo.data cfg/darknet_dynamic.cfg cfg/reweighting_net.cfg path/toweightfile python scripts/voc_eval.py results/path/to/comp4_det_test_” I don’t know the “path/toweightfile” and “results/path/to/comp4_det_test_” The weight file is your best weights saved, and the "results/path/to/comp4_det_test_" is your results containing boxes and label. when the first instruction executed successfully, and the the "results/path/to/comp4_det_test_" is generated. You should scan source code. — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe. Triage notifications on the go with GitHub Mobile for iOS or Android.
  
the
  
Is the first instruction written like this?&nbsp;python valid_ensemble.py cfg/metayolo.data cfg/darknet_dynamic.cfg cfg/reweighting_net.cfg backup/metayolo/000010.weights I want to know the best weights is darknet19_448.cfg or backup/metayolo/.weights? Because when I use darknet19_448.cfg the code can't run
 …
 ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: "bingykang/Fewshot_Detection" @.&gt;; 发送时间:&nbsp;2021年9月23日(星期四) 上午9:19 @.&gt;; @.@.**&gt;; 主题:&nbsp;Re: [bingykang/Fewshot_Detection] details about the base model and the fine-tuned (#18) Hi~How did you perform the code when evaluating the model？ “python valid_ensemble.py cfg/metayolo.data cfg/darknet_dynamic.cfg cfg/reweighting_net.cfg path/toweightfile python scripts/voc_eval.py results/path/to/comp4_det_test_” I don’t know the “path/toweightfile” and “results/path/to/comp4_det_test_” The weight file is your best weights saved, and the "results/path/to/comp4_det_test_" is your results containing boxes and label. when the first instruction executed successfully, and the the "results/path/to/comp4_det_test_" is generated. You should scan source code. — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe. Triage notifications on the go with GitHub Mobile for iOS or Android.
  
the instruction is to test, so it is no relation to  darknet19_448.cfg, the  darknet19_448.weights is pretrained weights to get backup/metayolo/000010.weights!!!
 
—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by futureisatyourhand** [comment]
> I want to know why the ***.txt is empty?It has no information.
> […](#)
> ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: "bingykang/Fewshot_Detection" ***@***.***&gt;; 发送时间:&nbsp;2021年9月23日(星期四) 中午11:40 ***@***.***&gt;; ***@***.******@***.***&gt;; 主题:&nbsp;Re: [bingykang/Fewshot_Detection] details about the base model and the fine-tuned (#18) Is the first instruction written like this?&nbsp;python valid_ensemble.py cfg/metayolo.data cfg/darknet_dynamic.cfg cfg/reweighting_net.cfg backup/metayolo/000010.weights I want to know the best weights is darknet19_448.cfg or backup/metayolo/.weights? Because when I use darknet19_448.cfg the code can't run … ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: "bingykang/Fewshot_Detection" @.&gt;; 发送时间:&nbsp;2021年9月23日(星期四) 上午9:19 @.&gt;; @.@.**&gt;; 主题:&nbsp;Re: [bingykang/Fewshot_Detection] details about the base model and the fine-tuned (#18) Hi~How did you perform the code when evaluating the model？ “python valid_ensemble.py cfg/metayolo.data cfg/darknet_dynamic.cfg cfg/reweighting_net.cfg path/toweightfile python scripts/voc_eval.py results/path/to/comp4_det_test_” I don’t know the “path/toweightfile” and “results/path/to/comp4_det_test_” The weight file is your best weights saved, and the "results/path/to/comp4_det_test_" is your results containing boxes and label. when the first instruction executed successfully, and the the "results/path/to/comp4_det_test_" is generated. You should scan source code. — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe. Triage notifications on the go with GitHub Mobile for iOS or Android. the Is the first instruction written like this?&nbsp;python valid_ensemble.py cfg/metayolo.data cfg/darknet_dynamic.cfg cfg/reweighting_net.cfg backup/metayolo/000010.weights I want to know the best weights is darknet19_448.cfg or backup/metayolo/.weights? Because when I use darknet19_448.cfg the code can't run … ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: "bingykang/Fewshot_Detection" @.&gt;; 发送时间:&nbsp;2021年9月23日(星期四) 上午9:19 @.&gt;; @.@.**&gt;; 主题:&nbsp;Re: [bingykang/Fewshot_Detection] details about the base model and the fine-tuned (#18) Hi~How did you perform the code when evaluating the model？ “python valid_ensemble.py cfg/metayolo.data cfg/darknet_dynamic.cfg cfg/reweighting_net.cfg path/toweightfile python scripts/voc_eval.py results/path/to/comp4_det_test_” I don’t know the “path/toweightfile” and “results/path/to/comp4_det_test_” The weight file is your best weights saved, and the "results/path/to/comp4_det_test_" is your results containing boxes and label. when the first instruction executed successfully, and the the "results/path/to/comp4_det_test_" is generated. You should scan source code. — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe. Triage notifications on the go with GitHub Mobile for iOS or Android. the instruction is to test, so it is no relation to darknet19_448.cfg, the darknet19_448.weights is pretrained weights to get backup/metayolo/000010.weights!!! — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe. Triage notifications on the go with GitHub Mobile for iOS or Android.

your result is none! You must look code.
